xarray/tests/__init__.py

```diff
@@ -182,6 +182,13 @@ def assertDataArrayAllClose(self, ar1, ar2, rtol=1e-05, atol=1e-08):
         assert_allclose(ar1, ar2, rtol=rtol, atol=atol)
 
 
+@contextmanager
+def raises_regex(error, pattern):
+    with pytest.raises(error) as excinfo:
+        yield
+    excinfo.match(pattern)
```

Comment:
Just to play devils advocate, 3.0 has been out for a year now. For us developers, its trivial to get the current release via conda/pip... so I'm not sure we really need to worry about maintaining backward compatibility. 

Obviously, I won't fight you if you think we should handle version 2 as well.